query/query0_test.go

```diff
@@ -3484,6 +3484,38 @@ func TestMatchingWithPagination(t *testing.T) {
 	}
 }
 
+func TestKRandomNodes(t *testing.T) {
+	q := `{
+		data(func: uid(61, 62, 63, 64, 65, 66, 67, 68, 69, 70), random: 2) @filter(has(connects)) {
+			kname
+			connects(random:2){
+				kname
+			}
+		}
+	}`
+	result := processQueryNoErr(t, q)
+	expected := `{"data":{"data":[{
+		"kname":"can_be_picked",
+		"connects":[
+			{"kname":"yes"},
+			{"kname":"yes"}
+			]},
+		{"kname":"can_be_picked",
+		"connects":[
+			{"kname":"yes"},
+			{"kname":"yes"}
+		]}]}}`
+	require.JSONEq(t, expected, result)
+
+	q = `{
+		data(func: uid(61, 62, 63, 64, 65, 66, 67, 68, 69, 70), random: 10) @filter(has(connects)) {
```

Comment:
What if number of input uids is less than 10?

Reply:
It was just by chance that, number of nodes in the query and the value of random was chosen to be the same. No specific reason.